zachsinterview/src/main/java/com/imprev/zachsinterview/NumericalKeyValueFun.java

```diff
@@ -0,0 +1,13 @@
+package com.imprev.zachsinterview;
+
+/**
+ * Provides some additional methods we may want to extend our NumericalKeyValueStore library with
+ */
+public interface NumericalKeyValueFun {
```

Comment:
Are these interfaces necessary, what are they adding? We don't implement each interface more than one time. Are we planning multiple implementations?